tests/testthat/test-weekly.R

```diff
@@ -0,0 +1,721 @@
+})
+
+
+test_that("estimate_R_agg works in parametric mode", {
+  ## estimate the reproduction number (method "parametric_si")
+  ## when not specifying t_start and t_end in config, they are set to estimate
+  ## the reproduction number on sliding weekly windows      
+  method <- "parametric_si"
+  config <- make_config(list(mean_si = mean_si,
+                             std_si = std_si))
+  res_weekly <- suppressWarnings(estimate_R_agg(incid = weekly_inc, 
+                        dt = 7L, # aggregation window of the data
+                        dt_out = 7L, # desired sliding window length
+                        iter = 10L,
+                        config = config,
+                        method = method,
+                        grid = list(precision = 0.001, min = -1, max = 1)))
+  
+  res_agg1 <- suppressWarnings(estimate_R_agg(incid = agg_inc, 
+                                              dt = c(2L,2L,3L), 
+                                              dt_out = 7L, 
+                                              iter = 10L,
+                                              config = config,
+                                              method = method,
+                                              grid = list(precision = 0.001, min = -1, max = 1)))
+  
+  res_agg2 <- suppressWarnings(estimate_R_agg(incid = agg_inc, 
+                                              dt = rep(c(2L,2L,3L), length.out=length(agg_inc)), 
+                                              dt_out = 7L, 
+                                              iter = 10L,
+                                              config = config,
+                                              method = method,
+                                              grid = list(precision = 0.001, min = -1, max = 1)))
+  
+  res_agg3 <- suppressWarnings(estimate_R_agg(incid = agg_inc_1, 
+                                              dt = c(2L,1L,3L), 
+                                              dt_out = 7L, 
+                                              iter = 10L,
+                                              config = config,
+                                              method = method,
+                                              grid = list(precision = 0.001, min = -1, max = 1)))
+  
+  
+  res_daily <- suppressWarnings(estimate_R(incid = incid, 
+                               config = config,
+                               method = method))
+  
+  ######################################################################
+  ## test that the weekly incidence matches the aggregated daily one
+  ## except for first time window where we impose I = 1 on first day
+  
+  for(i in seq_len(floor(length(res_daily$I) / dt))[-1])
+  {
+    expect_equal(sum(res_daily$I[i*dt+(1:dt)]), 
+                 sum(res_weekly$I[i*dt+(1:dt)]))
+  }
+  
+  for(i in seq_len(floor(length(res_daily$I) / dt))[-1])
+  {
+    expect_equal(sum(res_daily$I[i*dt+(1:dt)]), 
```

Comment:
@rebeccanash  I don't think that is what you want to do, i.e. I think you can only put one comparison of 2 objects in expect_equal, can you check and also I think you should add res_agg_3?